engine/include/cubos/engine/render/shadows/directional_caster.hpp

```diff
@@ -19,10 +23,52 @@ namespace cubos::engine
     {
         CUBOS_REFLECT;
 
+        static constexpr int MaxSplits = 10;
+
         cubos::engine::ShadowCaster baseSettings;
 
-        int numSplits;                     ///< Number of splits for PSSM/CSM.
-        std::vector<float> splitDistances; ///< Distance of each split from 0 to 1.
-        float maxDistance = 0;             ///< Max distance (if 0, uses camera zFar).
+        int configNumSplits = 3; ///< Target number of splits for PSSM/CSM.
+        int numSplits = 0;       ///< Current number of splits for PSSM/CSM.
+
+        std::vector<float> configSplitDistances; ///< Target distance of each split's near plane from 0 to 1 (relative
+                                                 ///< to maxDistance). If the amount of distances doesn't match the
+                                                 ///< number of splits, default values will be used.
+        std::vector<float> splitDistances;       ///< Distance of each split's near plane from 0 to 1.
+
```

Comment:
Also, isn't the `numSplits` variable redundant? Isn't it always equal to the size of splitDistances vector?